test/tests/functional/pbs_job_dependency.py

```diff
@@ -0,0 +1,287 @@
+        as it is submitted, then submit another job dependent on the
+        second held job and see if that gets held as well.
+        Also test that all jobs have dependency on other two jobs.
+        """
+        self.create_vnodes()
+        job = Job(TEST_USER)
+        j1 = self.server.submit(job)
+        a = {ATTR_state: 'R'}
+        self.server.expect(JOB, a, id=j1)
+
+        a = {ATTR_depend: 'runone:' + j1}
+        job2 = Job(TEST_USER, attrs=a)
+        j2 = self.server.submit(job2)
+        a = {ATTR_state: 'H', ATTR_h: 's'}
+        self.server.expect(JOB, a, id=j2)
+
+        a = {ATTR_depend: 'runone:' + j2}
+        job3 = Job(TEST_USER, attrs=a)
+        j3 = self.server.submit(job3)
+        a = {ATTR_state: 'H', ATTR_h: 's'}
+        self.server.expect(JOB, a, id=j3)
+
+        self.assert_dependency(j1, j2, j3)
+
+    def test_runone_depend_basic_on_job_array(self):
+        """
+        Test basic runone dependency tests on job arrays
+        1 - Submit a job array that runs and then submit a job having "runone"
+        dependency on the parent job array. Check that second job is deleted as
+        soon as it is submitted.
+        2 - Submit a can never run array job and submit a second job having
+        "runone" dependency on the array parent. Check that first job is
+        deleted when second job runs.
+        3 - Submit a runone dependency on an array subjob and check if job
+        submission fails in this case.
+        """
+
+        self.create_vnodes()
+        job = Job(attrs={'Resource_List.select': '1:ncpus=1',
+                         ATTR_J: '1-3'})
+        job.set_sleep_time(5)
+        j1 = self.server.submit(job)
+        d_job = Job(attrs={'Resource_List.select': '2:ncpus=1',
+                           ATTR_depend: 'runone:' + j1})
+        j1_2 = self.server.submit(d_job)
+        self.server.expect(JOB, {'job_state': 'B'}, id=j1)
+        self.server.expect(JOB, {'job_state': 'H'}, id=j1_2)
+        self.server.accounting_match("Job deleted as result of dependency",
+                                     id=j1_2)
+
+        job = Job(attrs={'Resource_List.select': '1:ncpus=4:NewRes=ver3',
+                         ATTR_J: '1-3'})
+        job.set_sleep_time(2)
+        j2 = self.server.submit(job)
+        d_job = Job(attrs={'Resource_List.select': '2:ncpus=1',
+                           ATTR_depend: 'runone:' + j2})
+        j2_1 = self.server.submit(d_job)
+        self.server.expect(JOB, {'job_state': 'R'}, id=j2_1)
+        self.server.expect(JOB, {'job_state': 'H'}, id=j2)
+        self.server.accounting_match("Job deleted as result of dependency",
+                                     id=j2)
+
+        job = Job(attrs={'Resource_List.select': '1:ncpus=1',
+                         ATTR_J: '1-3'})
+        job.set_sleep_time(2)
+        j3 = self.server.submit(job)
+        j3_1 = j3.split('[')[0] + '1' + j3.split('[')[1]
```

Comment:
There's a PTL function that will do this for you.  It's called create_subjob_id()